File: frontend/src/api/searchConsole.ts
import { api } from './client'

export type KeywordOpportunity = {
  keyword: string
  article_id: string | null
  article_title: string | null
  position: number | null
  clicks: number | null
  visits: number | null
  variation: number | null
  opportunity: string | null
  source: 'google_search_console'
}

export type KeywordOpportunitiesResponse = {
  connected: boolean
  keywords: KeywordOpportunity[]
  message: string | null
}

export function getKeywordOpportunities(projectId: string): Promise<KeywordOpportunitiesResponse> {
  return api.get<KeywordOpportunitiesResponse>(`/projects/${projectId}/search-console/keywords`)
}

export type SearchConsoleStatus = {
  connected: boolean
  provider?: string
  message: string | null
  docs_url?: string
}

export type SearchConsolePage = {
  url: string
  clicks: number
  impressions: number
  ctr: number
  position: number
}

export type SearchConsolePerformance = {
  connected?: boolean
  clicks: number
  impressions: number
  ctr: number
  position: number
  date?: string
  message?: string | null
}

export type SearchConsolePagesResponse = {
  connected: boolean
  pages: SearchConsolePage[]
  message: string | null
}

export function getSearchConsoleStatus(projectId: string): Promise<SearchConsoleStatus> {
  return api.get<SearchConsoleStatus>(`/projects/${projectId}/search-console/status`)
}

export function getSearchConsolePages(projectId: string): Promise<SearchConsolePagesResponse> {
  return api.get<SearchConsolePagesResponse>(`/projects/${projectId}/search-console/pages`)
}

export function getSearchConsolePerformance(projectId: string): Promise<SearchConsolePerformance> {
  return api.get<SearchConsolePerformance>(`/projects/${projectId}/search-console/performance`)
}

File: app/routers/search_console.py
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session
from app.core.database import get_db
from app.dependencies.auth import get_project_member
from app.models.project_member import ProjectMember
from app.models.project import Project

router = APIRouter(prefix="/projects/{project_id}/search-console", tags=["search_console"])


@router.get("/status")
def search_console_status(
    project_id: str,
    member: ProjectMember = Depends(get_project_member),
    db: Session = Depends(get_db),
):
    return {
        "connected": False,
        "provider": "google_search_console",
        "message": "Google Search Console nécessite une configuration OAuth. Consultez la documentation pour connecter votre compte Google.",
        "docs_url": "https://developers.google.com/webmaster-tools/search-console-api",
    }


@router.get("/keywords")
def search_console_keywords(
    project_id: str,
    member: ProjectMember = Depends(get_project_member),
    db: Session = Depends(get_db),
):
    return {
        "connected": False,
        "keywords": [],
        "message": "Google Search Console non connecté. Configurez l'intégration via OAuth pour accéder aux données de recherche.",
    }


@router.get("/pages")
def search_console_pages(
    project_id: str,
    member: ProjectMember = Depends(get_project_member),
    db: Session = Depends(get_db),
):
    return {
        "connected": False,
        "pages": [],
        "message": "Google Search Console non connecté.",
    }


@router.get("/performance")
def search_console_performance(
    project_id: str,
    member: ProjectMember = Depends(get_project_member),
    db: Session = Depends(get_db),
):
    return {
        "connected": False,
        "impressions": 0,
        "clicks": 0,
        "ctr": 0.0,
        "position": 0.0,
        "message": "Google Search Console non connecté.",
    }
